pkg/cmd/diff/cmd.go

```diff
@@ -0,0 +1,192 @@
+	"fmt"
+	"strings"
+	"sync"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2/registry"
+
+	"kitops/pkg/cmd/options"
+	"kitops/pkg/lib/constants"
+	"kitops/pkg/output"
+)
+
+const (
+	// Constants for identifying reference types
+	remotePrefix = "remote://"
+	localPrefix  = "local://"
+	//Constants for formatting output tables.
+	layerTableHeadings = "Type    | Digest             | Size"
+	layerTableFormat   = "%-7s | %-18s | %s\n"
+)
+
+type diffOptions struct {
+	options.NetworkOptions
+	configHome string
+	refA       *registry.Reference
+	refB       *registry.Reference
+}
+
+func DiffCommand() *cobra.Command {
+	opts := &diffOptions{}
+	cmd := &cobra.Command{
+		Use:   "diff",
+		Short: "Compare two ModelKits",
+		Args:  cobra.ExactArgs(2),
+		Long: `Compare two ModelKits to see the differences in their layers.
+		
+		ModelKits can be specified from either a local or from a remote registry.
+		To specify a local ModelKit, prefix the reference with 'local://', e.g. 'local://jozu.ml/foo/bar'.
+		To specify a remote ModelKit, prefix the reference with 'remote://', e.g. 'remote://jozu.ml/foo/bar'.
+		If no prefix is specified, the local registry will be checked first.
+		`,
+		Example: "kit diff modelkit1 modelkit2",
+		RunE:    runCommand(opts),
+	}
+	return cmd
+}
+
+func runCommand(opts *diffOptions) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := opts.complete(cmd.Context(), args); err != nil {
+			return output.Fatalf("Invalid arguments: %s", err)
+		}
+
+		var (
+			diffA, diffB *diffInfo
+			errA, errB   error
+		)
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		go func() {
+			defer wg.Done()
+			diffA, errA = getManifest(cmd.Context(), args[0], opts.refA, opts)
+		}()
+
+		go func() {
+			defer wg.Done()
+			diffB, errB = getManifest(cmd.Context(), args[1], opts.refB, opts)
+		}()
+
+		wg.Wait()
+
+		if errA != nil {
+			return output.Fatalf("Failed to get manifest for ModelKit1: %s", errA)
+		}
+		if errB != nil {
+			return output.Fatalf("Failed to get manifest for ModelKit2: %s", errB)
+		}
+
+		// Compare the two manifests
+		if diffA.Descriptor.Digest == diffB.Descriptor.Digest {
+			output.Infoln("ModelKits are identical")
+			return nil
+		}
+
+		result := compareManifests(diffA.Manifest, diffB.Manifest)
+		// Header
+		output.Infoln("Comparing:")
+		output.Infof("  ModelKit1: %s\n", opts.refA.String())
+		output.Infof("  ModelKit2: %s\n\n", opts.refB.String())
+
+		output.Infoln("Configurations:")
+		output.Infoln("---------------------------------------")
+		if result.SameConfig {
+			output.Infof("  Configs are identical (Digest: %s)\n\n", diffA.Manifest.Config.Digest[:17])
+
+		} else {
+			fmt.Printf("Configs differ:\n")
+			fmt.Printf("  ModelKit1 Config Digest: %s\n", diffA.Manifest.Config.Digest[:17])
+			fmt.Printf("  ModelKit2 Config Digest: %s\n\n", diffB.Manifest.Config.Digest[:17])
```

Comment:
I feel like it would be relatively easy to diff the configs themselves, but that can be a future task.

One note: since we're storing digests in the config for now, any difference in layers will also show up as a difference in configs, even if it's not intended (i.e. we're probably looking to compare Kitfiles here).